### ISSUE TITLE
Fix solc warnings

### ISSUE THREAD

**Issue by palango** [raiden-network/raiden-contracts#36]
When using the `ContractManager` to compile the onctracts I get shown the following warnings. These should be fixed.
```
Token.sol:6:5: Warning: Functions in interfaces should be declared external.
    function totalSupply() public constant returns (uint256 supply);
    ^--------------------------------------------------------------^
Token.sol:10:5: Warning: Functions in interfaces should be declared external.
    function balanceOf(address _owner) public constant returns (uint256 balance);
    ^---------------------------------------------------------------------------^
Token.sol:16:5: Warning: Functions in interfaces should be declared external.
    function transfer(address _to, uint256 _value) public returns (bool success);
    ^---------------------------------------------------------------------------^
Token.sol:23:5: Warning: Functions in interfaces should be declared external.
    function transferFrom(address _from, address _to, uint256 _value) public returns (bool success);
    ^----------------------------------------------------------------------------------------------^
Token.sol:29:5: Warning: Functions in interfaces should be declared external.
    function approve(address _spender, uint256 _value) public returns (bool success);
    ^-------------------------------------------------------------------------------^
Token.sol:34:5: Warning: Functions in interfaces should be declared external.
    function allowance(address _owner, address _spender) public constant returns (uint256 remaining);
    ^-----------------------------------------------------------------------------------------------^
TokenNetwork.sol:668:14: Warning: Use of the "var" keyword is deprecated.
        var (r, s, v) = signatureSplit(signature);
             ^
TokenNetwork.sol:668:17: Warning: Use of the "var" keyword is deprecated.
        var (r, s, v) = signatureSplit(signature);
                ^
TokenNetwork.sol:668:20: Warning: Use of the "var" keyword is deprecated.
        var (r, s, v) = signatureSplit(signature);
                   ^
Token.sol:6:5: Warning: Functions in interfaces should be declared external.
    function totalSupply() public constant returns (uint256 supply);
    ^--------------------------------------------------------------^
Token.sol:10:5: Warning: Functions in interfaces should be declared external.
    function balanceOf(address _owner) public constant returns (uint256 balance);
    ^---------------------------------------------------------------------------^
Token.sol:16:5: Warning: Functions in interfaces should be declared external.
    function transfer(address _to, uint256 _value) public returns (bool success);
    ^---------------------------------------------------------------------------^
Token.sol:23:5: Warning: Functions in interfaces should be declared external.
    function transferFrom(address _from, address _to, uint256 _value) public returns (bool success);
    ^----------------------------------------------------------------------------------------------^
Token.sol:29:5: Warning: Functions in interfaces should be declared external.
    function approve(address _spender, uint256 _value) public returns (bool success);
    ^-------------------------------------------------------------------------------^
Token.sol:34:5: Warning: Functions in interfaces should be declared external.
    function allowance(address _owner, address _spender) public constant returns (uint256 remaining);
    ^-----------------------------------------------------------------------------------------------^
TokenNetwork.sol:668:14: Warning: Use of the "var" keyword is deprecated.
        var (r, s, v) = signatureSplit(signature);
             ^
TokenNetwork.sol:668:17: Warning: Use of the "var" keyword is deprecated.
        var (r, s, v) = signatureSplit(signature);
                ^
TokenNetwork.sol:668:20: Warning: Use of the "var" keyword is deprecated.
        var (r, s, v) = signatureSplit(signature);
                   ^
Token.sol:6:5: Warning: Functions in interfaces should be declared external.
    function totalSupply() public constant returns (uint256 supply);
    ^--------------------------------------------------------------^
Token.sol:10:5: Warning: Functions in interfaces should be declared external.
    function balanceOf(address _owner) public constant returns (uint256 balance);
    ^---------------------------------------------------------------------------^
Token.sol:16:5: Warning: Functions in interfaces should be declared external.
    function transfer(address _to, uint256 _value) public returns (bool success);
    ^---------------------------------------------------------------------------^
Token.sol:23:5: Warning: Functions in interfaces should be declared external.
    function transferFrom(address _from, address _to, uint256 _value) public returns (bool success);
    ^----------------------------------------------------------------------------------------------^
Token.sol:29:5: Warning: Functions in interfaces should be declared external.
    function approve(address _spender, uint256 _value) public returns (bool success);
    ^-------------------------------------------------------------------------------^
Token.sol:34:5: Warning: Functions in interfaces should be declared external.
    function allowance(address _owner, address _spender) public constant returns (uint256 remaining);
    ^-----------------------------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:6:5: Warning: Functions in interfaces should be declared external.
    function totalSupply() public constant returns (uint256 supply);
    ^--------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:10:5: Warning: Functions in interfaces should be declared external.
    function balanceOf(address _owner) public constant returns (uint256 balance);
    ^---------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:16:5: Warning: Functions in interfaces should be declared external.
    function transfer(address _to, uint256 _value) public returns (bool success);
    ^---------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:23:5: Warning: Functions in interfaces should be declared external.
    function transferFrom(address _from, address _to, uint256 _value) public returns (bool success);
    ^----------------------------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:29:5: Warning: Functions in interfaces should be declared external.
    function approve(address _spender, uint256 _value) public returns (bool success);
    ^-------------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:34:5: Warning: Functions in interfaces should be declared external.
    function allowance(address _owner, address _spender) public constant returns (uint256 remaining);
    ^-----------------------------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:6:5: Warning: Functions in interfaces should be declared external.
    function totalSupply() public constant returns (uint256 supply);
    ^--------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:10:5: Warning: Functions in interfaces should be declared external.
    function balanceOf(address _owner) public constant returns (uint256 balance);
    ^---------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:16:5: Warning: Functions in interfaces should be declared external.
    function transfer(address _to, uint256 _value) public returns (bool success);
    ^---------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:23:5: Warning: Functions in interfaces should be declared external.
    function transferFrom(address _from, address _to, uint256 _value) public returns (bool success);
    ^----------------------------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:29:5: Warning: Functions in interfaces should be declared external.
    function approve(address _spender, uint256 _value) public returns (bool success);
    ^-------------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:34:5: Warning: Functions in interfaces should be declared external.
    function allowance(address _owner, address _spender) public constant returns (uint256 remaining);
    ^-----------------------------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:6:5: Warning: Functions in interfaces should be declared external.
    function totalSupply() public constant returns (uint256 supply);
    ^--------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:10:5: Warning: Functions in interfaces should be declared external.
    function balanceOf(address _owner) public constant returns (uint256 balance);
    ^---------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:16:5: Warning: Functions in interfaces should be declared external.
    function transfer(address _to, uint256 _value) public returns (bool success);
    ^---------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:23:5: Warning: Functions in interfaces should be declared external.
    function transferFrom(address _from, address _to, uint256 _value) public returns (bool success);
    ^----------------------------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:29:5: Warning: Functions in interfaces should be declared external.
    function approve(address _spender, uint256 _value) public returns (bool success);
    ^-------------------------------------------------------------------------------^
/Users/paul/Projects/raiden-contracts/raiden_contracts/contracts/Token.sol:34:5: Warning: Functions in interfaces should be declared external.
    function allowance(address _owner, address _spender) public constant returns (uint256 remaining);
    ^-----------------------------------------------------------------------------------------------^
```

**Comment by LefterisJP** [comment]
Yeah we will fix them don't worry. WE also got that in Raiden. They are harmless for now. What's more concerning to me is that we make changes in contracts of 2 different repos.

**Comment by LefterisJP** [comment]
We are waiting for solc v.0.4.22. Same in the main raiden repo.

Reason:

If you rename these to `external` now then the contracts which implement these interfaces use the public keyword and that is an error. With v0.4.22 [this patch](https://github.com/ethereum/solidity/pull/3681) will go in which will allow public to override external for interfaces.

**Comment by palango** [comment]
The last of these issues have been fixed in #57
